File: backend/src/models/Contenedor.js
const { DataTypes } = require('sequelize'); // Asegurar importación correcta
const sequelize = require('../config/dbConfig');
const Ruta = require('./Ruta');

const Contenedor = sequelize.define('Contenedor', {
  ubicacion: { type: DataTypes.STRING, allowNull: false },
  capacidad_max: { type: DataTypes.INTEGER, allowNull: false },
  nivel_llenado: { type: DataTypes.INTEGER, defaultValue: 0 },
  lat: { type: DataTypes.FLOAT }, // Añadido
  lng: { type: DataTypes.FLOAT }, // Añadido
  umbral_critico: { type: DataTypes.INTEGER, defaultValue: 80 }
},{
  tableName: 'contenedores', // Nombre de tabla explícito
  timestamps: true
});

module.exports = Contenedor; // Exportar correctamente

File: ecosmart_frontend/lib/models/contenedor.dart
class Contenedor {
  final int id;
  final String ubicacion;
  final int capacidadMax;
  final int nivelLlenado;
  final double? lat;
  final double? lng;
  final int umbralCritico;

  Contenedor({
    required this.id,
    required this.ubicacion,
    required this.capacidadMax,
    required this.nivelLlenado,
    this.lat,
    this.lng,
    required this.umbralCritico,
  });

  factory Contenedor.fromJson(Map<String, dynamic> json) {
    return Contenedor(
      id: json['id'],
      ubicacion: json['ubicacion'],
      capacidadMax: json['capacidad_max'],
      nivelLlenado: json['nivel_llenado'],
      lat: json['lat']?.toDouble(),
      lng: json['lng']?.toDouble(),
      umbralCritico: json['umbral_critico'],
    );
  }

  static List<Contenedor> fromList(List<dynamic> list) {
    return list.map((e) => Contenedor.fromJson(e)).toList();
  }
  
}